### PR TITLE
Allow repo names with ".git" in them

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph @allewun   implements #305 

* fixes a bug where repo names that contain ".git" are improperly handled when creating PRs. This happens if you want to create a PR for an Originate blog post. The repo name there is "originate.github.com".

* simplifies the repo specs. The repo specs always seemed over-engineered and cryptic to me. What does "HTTPS not ending with .git" mean? How exactly does the respective URL look like? Does this really cover all possible remote URLs? This over-optimization made the specs hard to read, and hard to maintain.

  I needed to add a different repo name to the specs, and that made the existing tables completely ridiculous. So I simplified them. Now we can give as many different origin URL as we want, even broken or strange ones, and make sure Git Town behaves correctly. The spec becomes more understandable for strangers, and I was able to remove a good amount of pretty complex test infrastructure.

What do you think?

### CODE REVIEW

File: features/step_definitions/pull_request_steps.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Then(/^I see a new (.+?) pull request for the "(.+?)" branch in the "(.+?)" repo in my browser$/) do |domain, branch, repo|
+  expect(@last_run_result.out).to eql "#{@tool} called with: #{pull_request_url domain, branch, repo}\n"
+end
```
Comment (charlierudolph) [inline features/step_definitions/pull_request_steps.rb]: To me, these all deal with the output of running a git-town command. The object to me in each case is the output (even though its not explicitly stated).

File: features/git-repo/bitbucket.feature [hunk 2]
```diff
@@ -1,16 +1,16 @@
 Feature: git-repo when origin is on Bitbucket
 
   Scenario Outline: result
-    Given my remote origin is on Bitbucket through <PROTOCOL>
+    Given my remote origin is <ORIGIN>
     And I have "open" installed
     When I run `git repo`
     Then I see the homepage of my Bitbucket repository in my browser
```
Comment (charlierudolph) [inline features/git-repo/bitbucket.feature]:
```
Then I see the Bitbucket homepage of the "<REPOSITORY>" repo in my browser.
```

I think the git-repo steps should also specify the repository similar to git-pr. Thus same for the github and multi platform support

File: features/step_definitions/pull_request_steps.rb (hunk 1 above)
Comment (kevgo) [inline features/step_definitions/pull_request_steps.rb]: Everything in the specs is about something that comes from running a git town command. So I don't think this way of looking at it is very helpful here. This is about seeing a pull request in the browser. `run_steps` are for things when the user runs something on the command line ("when I run 'git foo'...").

File: features/git-repo/bitbucket.feature (hunk 2 above)
Comment (kevgo) [inline features/git-repo/bitbucket.feature]: Good idea. Implemented.

File: features/step_definitions/pull_request_steps.rb (hunk 1 above)
Comment (charlierudolph) [inline features/step_definitions/pull_request_steps.rb]: I disagree with 
```
Everything in the specs is about something that comes from running a git town command
```
The specs verify two things. The state after the git town command runs and the output of the command. A lot of the specs are about verifying the state. The output of the command is limited to: what get printed to the terminal, opening a browser and opening a manpage.

`run_steps.rb` is not a very long file and I said in my initial comment at the moment is contains all references to @last_run_result which is used for verifying the output of the command. I am returning to my original suggestion and would like this moved back to `run_steps.rb`

Comment (kevgo) [inline features/step_definitions/pull_request_steps.rb]: I think we are talking past each other, and now this is going in circles. Let's clarify this offline.  :)

Comment (charlierudolph) [inline features/step_definitions/pull_request_steps.rb]: As per our discussion move this with the step about viewing a repository homepage into browser_helpers. Then :ship:

**Pull request closed without merging**